File: src/main/java/net/volcano/jdacommands/model/command/arguments/implementation/codec/BooleanCodec.java
package net.volcano.jdacommands.model.command.arguments.implementation.codec;

import net.volcano.jdacommands.model.ParameterData;
import net.volcano.jdacommands.model.command.arguments.CommandArgument;
import net.volcano.jdacommands.model.command.arguments.implementation.argument.BooleanArgument;
import net.volcano.jdacommands.model.command.arguments.interfaces.Codec;
import org.springframework.stereotype.Component;

@Component
public class BooleanCodec extends Codec<Boolean> {
	
	@Override
	protected CommandArgument<Boolean> buildArgument(ParameterData data) {
		var builder = BooleanArgument.builder();
		
		return builder.build();
	}
}

File: src/main/java/net/volcano/jdacommands/model/command/arguments/implementation/codec/BooleanCodec.kt
package net.volcano.jdacommands.model.command.arguments.implementation.codec

import net.volcano.jdacommands.model.ParameterData
import net.volcano.jdacommands.model.command.arguments.CommandArgument
import net.volcano.jdacommands.model.command.arguments.implementation.argument.BooleanArgument
import net.volcano.jdacommands.model.command.arguments.interfaces.Codec
import org.springframework.stereotype.Component

@Component
class BooleanCodec : Codec<Boolean?>() {

	override fun buildArgument(data: ParameterData): CommandArgument<Boolean?> {
		return BooleanArgument()
	}
}